cmd/osbuild-worker/jobimpl-bootc-manifest.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/osbuild/images/pkg/manifest"
+	"github.com/osbuild/osbuild-composer/internal/worker"
+	"github.com/osbuild/osbuild-composer/internal/worker/clienterrors"
+)
+
+type BootcManifestJobImpl struct{}
+
+type ErrorInvalidImageType string
+
+func (e ErrorInvalidImageType) Error() string { return string(e) }
+
+type ErrorCommandExecution string
+
+func (e ErrorCommandExecution) Error() string { return string(e) }
+
+type ErrorInvalidArguments string
+
+func (e ErrorInvalidArguments) Error() string { return string(e) }
+
+type ErrorUnknown string
+
+func (e ErrorUnknown) Error() string { return string(e) }
+
+func setResponseError(err error, result *worker.BootcManifestJobResult) {
+	switch e := err.(type) {
+	case ErrorInvalidImageType:
+		result.JobError = clienterrors.New(
+			clienterrors.ErrorInvalidImageType,
+			"Unsupported image type",
+			e.Error(),
+		)
+	case ErrorCommandExecution:
+		result.JobError = clienterrors.New(
+			clienterrors.ErrorBootcManifestExec,
+			"Error running bootc-image-builder command",
+			e.Error(),
+		)
+	default:
+		result.JobError = clienterrors.New(
+			clienterrors.ErrorInvalidManifestArgs,
+			"Invalid bootc parameters or parameter combination",
+			err.Error(),
+		)
+	}
+}
+
+func buildManifestCommand(args *worker.BootcManifestJob) *exec.Cmd {
+	baseArgs := []string{
+		"sudo", "podman", "run",
+		"--privileged",
+		"--rm",
+		"--pull=missing",
+		"--storage-driver=vfs",
+		"--cgroup-manager=cgroupfs",
+		"--runtime=runc",
+		"--security-opt=label=type:unconfined_t",
+		"quay.io/centos-bootc/bootc-image-builder:latest",
+		"manifest",
```

Comment:
Adding `--local`, and running `podman pull REF` before bib is the preferred way, see https://github.com/osbuild/bootc-image-builder/pull/423